### PR TITLE
Export PolygonHull and ConcaveHullOfPolygons in CAPI

### PR DESCRIPTION
**Pull request by pramsey**
- [x] expose in CAPI
- [x] documentation complete
- [x] unit tests of CAPI

### CODE REVIEW

File: capi/geos_c.h.in [hunk 1]
```diff
@@ -3461,6 +3475,66 @@ extern GEOSGeometry GEOS_DLL *GEOSConcaveHull(
     double ratio,
     unsigned int allowHoles);
 
+/**
+* Computes a boundary-respecting hull of a polygonal geometry,
+* with hull shape determined by a target parameter
+* specifying the fraction of the input vertices retained in the result.
+* Larger values compute less concave results.
```
Comment (brendan-ward) [inline capi/geos_c.h.in:3482]:
```suggestion
* Larger values produce less concave results.
```

File: capi/geos_c.h.in [hunk 2]
```diff
@@ -3461,6 +3475,66 @@ extern GEOSGeometry GEOS_DLL *GEOSConcaveHull(
...
+* A value of 1 produces the convex hull; a value of 0 produces the original geometry.
+* An outer hull is computed if the parameter is positive,
+* an inner hull is computed if it is negative.
+*
+* \param g the polygonal geometry to process
+* \param vertexNumFraction the target fraction of number of input vertices in result
```
Comment (brendan-ward) [inline capi/geos_c.h.in:3488]:
```suggestion
* \param vertexNumFraction the target fraction of the count of input vertices to retain in result
```

File: capi/geos_c.h.in [hunk 3]
```diff
@@ -3461,6 +3475,66 @@ extern GEOSGeometry GEOS_DLL *GEOSConcaveHull(
...
+* A given set of polygons has a sequence of hulls of increasing concaveness,
+* determined by a numeric target parameter.
+* The computed hull "fills the gap" between the polygons,
+* and does not intersect their interior.
+*
+* The concave hull is constructed by removing the longest outer edges
+* of the Delaunay Triangulation of the space between the polygons,
+* until the target criterion parameter is reached.
+*
+* "Maximum Edge Length" is the length of the longest edge between the polygons is no larger than this value.
```
Comment (brendan-ward) [inline capi/geos_c.h.in:3512]:
```suggestion
* "Maximum Edge Length" constrains the length of the longest edge between the polygons to be no larger than this value.
```

File: capi/geos_c.h.in [hunk 4]
```diff
@@ -3461,6 +3475,66 @@ extern GEOSGeometry GEOS_DLL *GEOSConcaveHull(
...
+*
+* \param g the valid multi-polygonal geometry to process
+* \param lengthRatio determine the Maximum Edge Length as a
+*        fraction of the difference between the longest and
+*        shortest edge lengths between the polygons.
+*        This normalizes the Maximum Edge Length to be scale-free.
+*        A value of 1 produces the convex hull; a value of 0 produces
+*        the original polygons.
+* \param isHollesAllowed is the concave hull can be allowed to contain holes?
```
Comment (brendan-ward) [inline capi/geos_c.h.in]:
```suggestion
* \param isHollesAllowed is the concave hull allowed to contain holes?
```

File: capi/geos_c.h.in [hunk 5]
```diff
@@ -3461,6 +3475,66 @@ extern GEOSGeometry GEOS_DLL *GEOSConcaveHull(
...
+* The computed hull "fills the gap" between the polygons,
+* and does not intersect their interior.
+*
+* The concave hull is constructed by removing the longest outer edges
+* of the Delaunay Triangulation of the space between the polygons,
+* until the target criterion parameter is reached.
+*
+* "Maximum Edge Length" is the length of the longest edge between the polygons is no larger than this value.
+*
+* \param g the valid multi-polygonal geometry to process
```
Comment (brendan-ward) [inline capi/geos_c.h.in:3514]:
```suggestion
* \param g the valid MultiPolygon geometry to process
```

**Review by brendan-ward — COMMENTED** [review]
A few minor text edits to consider

File: capi/geos_c.h.in (hunk 5 above)
Comment (dr-jts) [reply capi/geos_c.h.in]: except that this accepts Polygons as well.  Should probably be "polygonal".  (Grrr that OGC SFS has such a awkward geometry class hierarchy)

**Comment by dr-jts** [comment]
I'm starting to wonder if `PolygonHull` should be renamed to `PolygonSimplifier`. Reasons are:
* there's perhaps too many "Hull" concepts in play right now?
* PolygonHull (outer and inner) essentially is similar to Visvalingham-Wyattt simplification, but ensuring topology and containment are enforced.
* `PolygonHull` preserves the number of input polygons, whereas `ConcaveHullOfPolygons` does not  
* it's possibe to extend the algorithm to produce a result which is simplified but not strictly inner or outer. This would be essentially "Topology-Preserving VW Simplification".   

Comments welcome.

If this change is made, it will be done in JTS as well.  It involves renaming the class, and probably moving it to the `simplify` package.

**Comment by pramsey** [comment]
I'm fine w/ renaming if that's your Zen. Make it so sooner than later so the train keeps on running!

**Comment by dr-jts** [comment]
JTS has renamed `PolygonHull` to `PolygonHullSimplifier` ([this PR](https://github.com/locationtech/jts/pull/880))

**Comment by pramsey** [comment]
```
extern GEOSGeometry GEOS_DLL *GEOSPolygonHullSimplify(
    const GEOSGeometry* g,
    unsigned int isOuter,
    double vertexNumFraction);
```
and 
```
extern GEOSGeometry GEOS_DLL *GEOSConcaveHullOfPolygons(
    const GEOSGeometry* g,
    double lengthRatio,
    unsigned int isTight,
    unsigned int isHolesAllowed);
```

**Pull request merged**